project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,41 @@
+from odoo import fields, models, api
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        string="PR State",
+        )
+
+    @api.model
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ('open', 'Open'),
+            ('draft', 'Draft'),
+            ('merged', 'Merged'),
+            ('closed', 'Closed'),
+        ]
+
+    @api.model
+    def create(self, vals):
+        if 'project_id' in vals and 'pr_uri' in vals:
```

Comment:
Why aren't you using the same function for both write and create?
It will simplify the flow in case it is needed to be extended later.
![image](https://user-images.githubusercontent.com/2234999/215358976-eba6917d-1fe2-4bc6-b127-1cf0cf7f5e82.png)